lib/saml20.js

```diff
@@ -97,10 +130,15 @@ exports.create = function(options, callback) {
       // </saml:Attribute>
       var attributeElement = doc.createElementNS(NAMESPACE, 'saml:Attribute');
       attributeElement.setAttribute('Name', prop);
+
+      if (options.includeAttributeNameFormat){
+        attributeElement.setAttribute('NameFormat', getNameFormat(prop));        
```

Comment:
prop cannot be undefined. If the json has a prop named undefined, it will be returned as a string in `Object.keys(json)`.